cpp/periodic_module/periodic_iterate.cpp

```diff
@@ -0,0 +1,241 @@
+    match_by_id_vector.push_back(fmt::format("MATCH ()-[{}]->() WHERE ID({}) = __{}_id", rel_name, rel_name, rel_name));
+  }
+
+  if (match_by_id_vector.size()) {
+    match_string = boost::algorithm::join(match_by_id_vector, " ");
+  }
+
+  return fmt::format("{} {} {}", unwind_batch, with_variables, match_string);
+}
+
+mg::Map ConstructParams(std::vector<std::string> columns, std::vector<std::vector<mg::Value>> &batch) {
+  auto params = mg::Map(1);
+  auto list_value = mg::List(batch.size());
+
+  auto param_row_size = columns.size();
+
+  for (size_t row = 0; row < batch.size(); row++) {
+    mg::Map constructed_row(param_row_size);
+
+    for (size_t i = 0; i < param_row_size; i++) {
+      if (batch[row][i].type() == mg::Value::Type::Node) {
+        constructed_row.Insert(columns[i], mg::Value(static_cast<int64_t>(batch[row][i].ValueNode().id().AsInt())));
+      } else if (batch[row][i].type() == mg::Value::Type::Relationship) {
+        constructed_row.Insert(columns[i],
+                               mg::Value(static_cast<int64_t>(batch[row][i].ValueRelationship().id().AsInt())));
+      } else {
+        constructed_row.Insert(columns[i], batch[row][i]);
+      }
+    }
+
+    list_value.Append(mg::Value(std::move(constructed_row)));
+  }
+
+  params.Insert(kBatchInternalName, mg::Value(std::move(list_value)));
+
+  return params;
+}
+
+std::string ConstructFinalQuery(std::string running_query, std::string preffix_query) {
+  return fmt::format("{} {}", preffix_query, running_query);
+}
+
+void ExecuteRunningQuery(std::string running_query, std::vector<std::string> columns,
+                         std::vector<std::vector<mg::Value>> &batch) {
+  if (!batch.size()) {
+    return;
+  }
+
+  auto param_names = ExtractParamNames(columns, batch[0]);
+  auto query_columns = ConstructColumns(columns, batch[0]);
+
+  auto preffix_query = ConstructQueryPreffix(param_names);
+  auto final_query = ConstructFinalQuery(running_query, preffix_query);
+
+  auto query_params = ConstructParams(columns, batch);
+
+  mg::Client::Params session_params{.host = "localhost", .port = 7687};
+  auto client = mg::Client::Connect(session_params);
+  if (!client) {
+    throw std::runtime_error("Unable to connect to client!");
+  }
+  if (!client->Execute(final_query, query_params.AsConstMap())) {
+    throw std::runtime_error("Error while executing periodic iterate!");
+  }
+
+  client->DiscardAll();
+}
+
+void PeriodicIterate(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  auto record = record_factory.NewRecord();
+
+  const auto input_query = std::string(arguments[0].ValueString());
+  const auto running_query = std::string(arguments[1].ValueString());
+  const auto config = arguments[2].ValueMap();
+
+  const auto batch_size_value = config.At(kConfigKeyBatchSize);
```

Comment:
`At` can throw and in this case it is not covered by any `try`. I would rather check with `contains` if `kConfigKeyBatchSize` is inside and throw something more with graceful message